docs/developer.md

```diff
@@ -0,0 +1,45 @@
+# Developer Documentation
+
+This documentation will assist developers trying to understand the working of the Gradle Build Server for easier
+contribution or integration.
+
+## Composite Builds
+
+### Challenge
+
+When using a model builder to fetch the GradleSourceSet model, the complete hierarchical scope of the project is lost.
+This hinders the ability to fetch included builds and project dependencies, which is crucial for composite builds.
+
+### Our Solution: Using Build Actions
+
+To address this limitation we take the following approach:
+
+1. We utilize the `BuildAction` to capture the complete hierarchical scope of the project, allowing us to retrieve all
+   the included builds and their associated information.
+2. After obtaining the included builds, we use the model builder to get source sets for each individual project.
+3. With te complete set of source sets for each included build we map the build target dependencies and return the list
+   of `GradleSourceSets`.
+
+![Composite Build](./images/composite_build.png)
+
+## Java Home Handling
+
+The default `JAVA_HOME` configuration of a project maybe incompatible with Gradle Build Server. If this happens we
+implemented a fallback logic to find a compatible JDK to launch the Gradle Build Server.
+
+1. We probe build the project and look for any `JAVA_HOME` incompatibility error.
```

Comment:
Give a simple and brief explanation of how the probe build is done:
1. what task is triggered as probe
2. how is java incompatibility error is determined

Reply:
Okay